plugins/nf-google/src/main/nextflow/cloud/google/batch/GoogleBatchTaskHandler.groovy

```diff
@@ -224,8 +224,14 @@ class GoogleBatchTaskHandler extends TaskHandler implements FusionAwareTask {
         if( executor.config.cpuPlatform )
             instancePolicy.setMinCpuPlatform( executor.config.cpuPlatform )
 
-        if( task.config.getMachineType() )
-            instancePolicy.setMachineType( task.config.getMachineType() )
+        def machineType = task.config.getMachineType()
+        def instanceTemplate = null
+        if( machineType ) {
+            if( machineType.startsWith('template://') )
+                instanceTemplate = machineType.minus('template://')
+            else
+                instancePolicy.setMachineType( machineType )
+        }
```

Comment:
This code it's becoming a bit convoluted. Let's add some descriptive comment to help to follow all logic here

Reply:
I moved the instance policy / instance template pieces into separate branches so that it's clearer